stat/samplemv/halton.go

```diff
@@ -0,0 +1,166 @@
+package samplemv
+
+import (
+	"fmt"
+	"math/rand"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat/distmv"
+)
+
+// Haltoner is a wrapper around the Halton sampling generation method.
+type Haltoner struct {
+	Kind HaltonKind
+	Q    distmv.Quantiler
+	Src  *rand.Rand
+}
+
+func (h Haltoner) Sample(batch *mat.Dense) {
+	Halton(batch, h.Kind, h.Q, h.Src)
+}
+
+type HaltonKind int
+
+const (
+	// HaltonOwen generates (scrambled) Halton samples using the Randomized van der Corput
+	// algorithm described in
+	//  A randomized Halton algorithm
+	//  Art Owen
+	//  https://arxiv.org/pdf/1706.02808.pdf
+	// Currently limited to 1000 dimensional inputs.
+	Owen = iota + 1
+)
+
+// Halton generates rows(batch) samples using Halton sequence from
```

Comment:
Should "Halton sequence" be plural? It seems so from usage in the paper.

Reply:
I don't know. The paper is about a generic algorithm over all Halton sequences. The wikipedia is similar. The wikipedia then talks about "let's take one dimension of the Halton sequence to be based on 2 and the other on 3". Similarly, Owen's paper says "The Halton sequence has form [blas] for different bases ... the discrepancy of the Halton sequence is [blah]".

I took this to mean that a single Halton sequence is for a specific choice of primes, and we don't give a choice of primes, so we use a specific Halton sequence. Not sure that I'm right though.

Reply:
I searched through the Owen paper and the vast majority of uses were a noun with an article.

Reply:
"The" added